### ISSUE TITLE
SR-GAN: why use detach while calculating content loss when the VGG model weights are already frozen

### ISSUE THREAD

**Issue by kushalchordiya216** [eriklindernoren/PyTorch-GAN#94]
The code snippet for content loss is as
`# Content loss
        gen_features = feature_extractor(gen_hr)
        real_features = feature_extractor(imgs_hr)
        loss_content = criterion_content(gen_features, real_features.detach())`
i dont understand why you've used detach (and also why its used specifically only on real_features), when the feature_extractor model already has all weights frozen

**Comment by pbdahzou** [comment]
@kushalchordiya216 hi，do you understand why using detach for real_features now? I also have the same question.

**Comment by kushalchordiya216** [comment]
Hi, i'm still not entirely sure but i have found experimentally, that it produces better results when trained that way, good luck with your efforts

**Comment by pbdahzou** [comment]
@kushalchordiya216  thank you very much for your quick reply，your experimental results are very useful for me！
